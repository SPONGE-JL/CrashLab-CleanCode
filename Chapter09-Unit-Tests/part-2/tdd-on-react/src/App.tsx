import React from 'react';

function App() {
  return (
    <div className="App">
      <h1>TDD on React.js</h1>
    </div>
  );
}

export default App;
