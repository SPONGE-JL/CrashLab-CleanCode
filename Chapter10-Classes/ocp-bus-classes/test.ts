import { DockingTrailerBus } from "./DockingTrailerBus";
import { DoubleDeckBus } from "./DoubleDeckBus";

const doubleDeckBus = new DoubleDeckBus({
  name: "이층버스",
  capacity: 60,
  maxDeck: 2
})

console.log(doubleDeckBus.getData());

const dockingTrailerBus = new DockingTrailerBus({
  name: "트레일러 버스",
  capacity: 100,
  dock: "클로이 후크형"
})

console.log(dockingTrailerBus.getData());