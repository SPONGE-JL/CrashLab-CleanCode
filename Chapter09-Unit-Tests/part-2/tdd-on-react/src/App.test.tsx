import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import 'jest-styled-components';

import { pickInputTagFrom } from 'Components/Input/index.test';

import App from './App';

describe('<App />', () => {
  it('should render its components.', () => {
    const { container } = render(<App />);

    // Exist Input Box
    const input = pickInputTagFrom(container, '');
    expect(input).toBeInTheDocument();
    const addButton = screen.getByText('Add').parentElement as HTMLElement;
    expect(addButton).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should handle state: "todo" - it makes the input tag clean after click the "Add button".', () => {
    const { container } = render(<App />);

    // Typing in input tag
    const input = pickInputTagFrom(container, '');
    fireEvent.change(input, { target: { value: 'type something' } });
    expect(input.value).toBe('type something');
    expect(container).toMatchSnapshot();

    // Click the Add Button
    const addButton = screen.getByText('Add').parentElement as HTMLElement;
    fireEvent.click(addButton);
    expect(input.value).toBe('');
    expect(container).toMatchSnapshot();
  });
});
