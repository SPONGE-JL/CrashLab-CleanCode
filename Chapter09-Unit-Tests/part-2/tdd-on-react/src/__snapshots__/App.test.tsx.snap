// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should handle state: "todo" - it makes the input tag clean after click the "Add button". 1`] = `
.c3 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

.c4 {
  text-align: center;
  cursor: pointer;
  padding: 10px 20px;
  border-radius: 8px;
  background-color: #304FFE;
}

.c4:hover {
  background-color: #1639FE;
}

.c4:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c5 {
  color: #FFFFFF;
  font-size: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  margin: 0% 25%;
  padding: 20px;
  border-radius: 8px;
  background-color: :#FFFFFF;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        cache
      </p>
      <div
        class="c2"
      >
        <input
          class="c3"
          placeholder="Insert a new todo"
          value="type something"
        />
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            Add
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> should handle state: "todo" - it makes the input tag clean after click the "Add button". 2`] = `
<div>
  <div
    class="sc-eCImPb eZMCne"
  >
    <div
      class="sc-jRQBWg itwRPG"
    >
      <p>
        type something
      </p>
      <div
        class="sc-hKwDye kIiDBr"
      >
        <input
          class="sc-bdvvtL lMNJP"
          placeholder="Insert a new todo"
          value=""
        />
        <div
          class="sc-gsDKAQ gSWOZW"
        >
          <div
            class="sc-dkPtRN kvylGA"
          >
            Add
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<App /> should render its components. 1`] = `
.c3 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

.c4 {
  text-align: center;
  cursor: pointer;
  padding: 10px 20px;
  border-radius: 8px;
  background-color: #304FFE;
}

.c4:hover {
  background-color: #1639FE;
}

.c4:active {
  box-shadow: inset 5px 5px 10px rgba(0,0,0,0.2);
}

.c5 {
  color: #FFFFFF;
  font-size: 16px;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 50%;
  margin: 0% 25%;
  padding: 20px;
  border-radius: 8px;
  background-color: :#FFFFFF;
  box-shadow: 5px 5px 10px rgba(0,0,0,0.2);
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <p>
        cache
      </p>
      <div
        class="c2"
      >
        <input
          class="c3"
          placeholder="Insert a new todo"
          value=""
        />
        <div
          class="c4"
        >
          <div
            class="c5"
          >
            Add
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Input /> should render with a props: "onChange" 1`] = `
.c0 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

<div>
  <input
    class="c0"
    value=""
  />
</div>
`;

exports[`<Input /> should render with a props: "onChange" 2`] = `
<input
  class="sc-bdvvtL lMNJP"
  value=""
/>
`;

exports[`<Input /> should render with a props: "placeholder" 1`] = `
.c0 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

<div>
  <input
    class="c0"
    placeholder="This is an Input tag."
    value=""
  />
</div>
`;

exports[`<Input /> should render with a props: "placeholder" 2`] = `
<input
  class="sc-bdvvtL lMNJP"
  placeholder="This is an Input tag."
  value=""
/>
`;

exports[`<Input /> should render with a props: "value" 1`] = `
.c0 {
  font-size: 16px;
  text-align: center;
  width: 100%;
  border-radius: 8px;
  border: 1px solid #BDBDBD;
  outline: none;
  padding: 10px;
  margin-right: 10px;
}

<div>
  <input
    class="c0"
    value="Inserted some text."
  />
</div>
`;

exports[`<Input /> should render with a props: "value" 2`] = `
<input
  class="sc-bdvvtL lMNJP"
  value="Inserted some text."
/>
`;
